core/services/feeds/service.go

```diff
@@ -178,6 +192,56 @@ func (s *service) GetJobProposal(id int64) (*JobProposal, error) {
 	return s.orm.GetJobProposal(context.Background(), id)
 }
 
+func (s *service) ApproveJobProposal(ctx context.Context, id int64) error {
+	if s.fmsClient == nil {
+		return errors.New("fms rpc client is not connected")
+	}
+
+	jp, err := s.orm.GetJobProposal(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	if jp.Status != JobProposalStatusPending {
+		return errors.New("must be a pending job proposal")
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, postgres.DefaultQueryTimeout)
+	defer cancel()
+
+	j, err := s.generateJob(jp.Spec)
+	if err != nil {
+		return err
+	}
+
+	err = s.txm.TransactWithContext(ctx, func(ctx context.Context) error {
+		// Create the job
+		_, err = s.jobSpawner.CreateJob(ctx, *j, j.Name)
+		if err != nil {
+			return err
+		}
+
+		// Approve the job
+		if err = s.orm.ApproveJobProposal(ctx, id, j.ExternalJobID, JobProposalStatusApproved); err != nil {
+			return err
+		}
+
+		// Send to FMS Client
+		if _, err = s.fmsClient.ApprovedJob(ctx, &pb.ApprovedJobRequest{
```

Comment:
Does this call an HTTP request?

Reply:
This is an RPC call

Reply:
It doesn't make sense to issue the RPC call inside the transaction. It ought to come either before or after.

Reply:
Yes because ee want to ensure that FMS knows when the job gets created, so we rollback the transaction if it fails to notify the FMS. The node op will be able to approve the transaction again once the RPC connection is reestablished.